soccer/src/soccer/strategy/agent/position/runner.hpp

```diff
@@ -0,0 +1,102 @@
+#pragma once
+
+#include <cmath>
+
+#include <rclcpp/rclcpp.hpp>
+#include <rclcpp_action/rclcpp_action.hpp>
+#include <spdlog/spdlog.h>
+
+#include <rj_msgs/action/robot_move.hpp>
+
+#include "planning/instant.hpp"
+#include "position.hpp"
+#include "rj_common/time.hpp"
+#include "rj_geometry/geometry_conversions.hpp"
+#include "rj_geometry/point.hpp"
+
+namespace strategy {
+
+/*
+ * The Offense position handles general offensive tasks, like passing to
+ * teammates, moving to get open, or shooting on goal.
+ */
+class Runner : public Position {
+public:
+    Runner(int r_id);
+    ~Runner() override = default;
+
+    void receive_communication_response(communication::AgentPosResponseWrapper response){};
+    communication::PosAgentResponseWrapper receive_communication_request(
+        communication::AgentPosRequestWrapper request){};
+
+    void derived_acknowledge_pass(){};
+    void derived_pass_ball(){};
+    void derived_acknowledge_ball_in_transit(){};
+
+private:
+    bool kicking_{true};
+
+    std::optional<RobotIntent> derived_get_task(RobotIntent intent) override;
+    // TODO (Kevin): strategy design pattern for BallHandler/Receiver
```

Comment:
LOL make sure to check what you're copying. not sure if you need Kevin to work on this